staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -573,6 +574,7 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 				handler = restfulGetResource(getter, exporter, reqScope)
 			}
 			handler = metrics.InstrumentRouteFunc(action.Verb, resource, subresource, handler)
+			handler = genericfilters.RestfulWithCompression(handler, a.group.Context)
```

Comment:
@ilackarms a [feature gate](https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go#L117) that defaults on would probably be best.